### PR TITLE
Add missing key props to FormFieldModal component

### PR DESCRIPTION
**Pull request by samuel-knutson**
The `FormFieldModal` component did not set proper keys. This caused console warnings as well as erratic runtime behavior. When I fixed that issue, I noticed that the form layout would sometimes get messed up when blurring a lookup field. It appears the form was making room for a hidden field. To resolve that issue, I process field resolvers before form layout, omit hidden fields from the form, and sort the fields to ensure a consistent ordering.

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -35,6 +35,10 @@ export function buildVariableSizeFieldSection(fields) {
   let currentRowWidth = 0;
 
   fields.forEach((field) => {
+    if (field.meta?.hidden) {
+      return;
+    }
+
```
Comment (ydsood) [inline src/util.js:41]: Is this related to the change?

File: src/util.js [hunk 2]
```diff
@@ -31,10 +31,11 @@ export function chunkConditional(array, size, predicate) {
 }
 
 export function buildVariableSizeFieldSection(fields) {
+  const sortedFields = [...fields].sort();
```
Comment (ydsood) [inline src/util.js:34]: callback for sort() function needs to added here

File: src/util.js (hunk 1 above)
Comment (samuel-knutson) [reply src/util.js:42]: Yes, see original comment:

> When I fixed that issue, I noticed that the form layout would sometimes get messed up when blurring a lookup field. It appears the form was making room for a hidden field. To resolve that issue, I process field resolvers before form layout, **omit hidden fields from the form**, and sort the fields to ensure a consistent ordering.

File: src/util.js (hunk 2 above)
Comment (samuel-knutson) [reply src/util.js:34]: oops 🤠
